src/Microsoft.ML.FastTree/FastTree.cs

```diff
@@ -3117,6 +3117,10 @@ private protected virtual bool SaveAsOnnx(OnnxContext ctx, string[] outputNames,
         }
         bool ISingleCanSaveOnnx.SaveAsOnnx(OnnxContext ctx, string[] outputNames, string featureColumn)
         {
+            const int minimumOpSetVersion = 9;
+            if (ctx.GetOpSetVersion() < minimumOpSetVersion)
+                throw Contracts.ExceptParam(nameof(minimumOpSetVersion), $"Requested OpSet version {ctx.GetOpSetVersion()} is lower than FastTree's minimum OpSet version requirement: {minimumOpSetVersion}");
+
             return SaveAsOnnx(ctx,outputNames,featureColumn);
```

Comment:
This `ISingleCanSaveOnnx.SaveAsOnnx` returns by calling `private protected virtual bool SaveAsOnnx` defined on this same class (`TreeEnsembleModelParameters`). I think it's better to move this exception you've created here into that other `private protected virtual` method, since there is the actual logic that uses onnx operators, and also because that other method can be called while exporting to onnx other classes (such as `FastTreeTweedieModelParameters` or `FastForestRegressionModelParameters`).

Also, I'd change the exception message from `is lower than FastTree's minimum OpSet` to `is lower than TreeEnsembleModelParameters`, since the `TreeEnsembleModelParameters` class is used by other models (not only FastTree) such as LightGBM... and I _think_ that when exporting to Onnx those other models this `SaveAsOnnx` method would also be called.

Reply:
fixed